runtime-transaction/src/runtime_transaction.rs

```diff
@@ -74,6 +77,24 @@ impl RuntimeTransaction<SanitizedVersionedTransaction> {
             .unwrap_or_else(|| is_simple_vote_transaction(&sanitized_versioned_tx));
         let message_hash =
             message_hash.unwrap_or_else(|| sanitized_versioned_tx.get_message().message.hash());
+
+        let precompile_signature_details = get_precompile_signature_details(
+            sanitized_versioned_tx
+                .get_message()
+                .program_instructions_iter()
+                .map(|(program_id, ix)| (program_id, SVMInstruction::from(ix))),
+        );
```

Comment:
This call does same instructions iteration as `ComputeBudgetInstructionDetails::try_from(...)` call below. Early we talked about to neatly iterate once to collect all "instruction internal details" that includes signatures and CB and builtins. I still want to do that, a little bit of performance and a much neater code. wdyt?

Reply:
If you also want to do that, can start from this pr :) Or can leave it as is, but combining ix iterations later. I am fine with both.

Reply:
I still want to get there, but think the best approach is incremental.

Right now we have a slower version of this code running in several places to get signature details, repeatedly.
Here's how I envision it:

- initial refactor lets us use this more efficient version
- transition to runtime transaction (use cached version)
- once fully transitioned to runtime transaction, we can implement the single-pass scan for all static meta

wdyt?